duration timeout
duration duration
string[] regex_end_condition
